src/checks.coffee

```diff
@@ -114,29 +114,28 @@ exports.json_array_brackets = (rules) ->
   for i in [ 'CMD', 'ENTRYPOINT', 'RUN', 'VOLUME' ]
     rule = this.getAll(i, rules)
     for r in rule
-      # First make sure we're actually dealing with the exec form
-      unless r.arguments[0].match(/\[|\]/g)
+      # First make sure we're actually dealing with the exec form (not ignoring position)
+      unless r.arguments[0].match(/(^\s*\[)|(\]\s*$)/g)
         continue
-
-      # First check if there is only one of each to begin with (ignore position).
-      lbracket = r.arguments[0].match(/\[/g)
-      if lbracket? && lbracket.length > 1
-        utils.log 'ERROR', "Multiple opening brackets found on line #{r.line}"
-        return 'failed'
-
-      rbracket = r.arguments[0].match(/\]/g)
-      if rbracket? && rbracket.length > 1
-        utils.log 'ERROR', "Multiple closing brackets found on line #{r.line}"
-        return 'failed'
-
-      # And finally verify the bracket's position
-      unless r.arguments[0].match(/^\[/)
-        utils.log 'ERROR', "No opening bracket found on line #{r.line}"
-        return 'failed'
-      unless r.arguments[0].match(/\]$/)
-        utils.log 'ERROR', "No closing bracket found on line #{r.line}"
+  
+      # Utility funtion from http://stackoverflow.com/a/16608045/4126114
+      isArray = (a) -> return (!!a) && (a.constructor is Array)
+  
+      # Check if this is a valid JSON array
+      try 
```

Comment:
Spacing in this block diverges from the rest of the codebase, could you please add whitespace around the `>` and `=` operators?